packages/signature-controller/src/types.ts

```diff
@@ -71,15 +77,41 @@ export type MessageParamsTyped = MessageParams & {
         primaryType: string;
         message: Json;
       };
-
   /** Version of the signTypedData request. */
   version?: string;
 };
 
+/** Information about various state changes returned by decoding api. */
+type DecodingDataStateChanges = {
```

Comment:
Rather than defining this as an array type, is it more flexible to define a singular type such as `DecodingDataStateChange` and then set the `stateChanges` property as an array of it?

Reply:
done